R/transformCounts.R

```diff
@@ -308,88 +303,75 @@ setMethod("relAbundanceCounts",signature = c(x = "SummarizedExperiment"),
              "Try log10 with pseudocount = 1 or other numeric value.",
              call. = FALSE)
     }
-
     mat <- log10(mat)
-
     return(mat)
 }
 
-.calc_pa <- function(mat, threshold){
+.calc_pa <- function(mat, threshold, ...){
     # If value is over zero, gets value 1. If value is zero, gets value 0.
     mat <- (mat > threshold) - 0L
     return(mat)
 }
 
-.calc_hellinger <- function(mat, threshold){
-
+.calc_hellinger <- function(mat, pseudocount, ...){
+    # Gets the relative abundance
+    mat <- .calc_rel_abund(mat)
+    mat <- .apply_pseudocount(mat, pseudocount)
     # If there is negative values, gives an error.
     if (any(mat < 0, na.rm = TRUE)) {
         stop("Abundance table contains negative values and hellinger ",
-             "transformation is being applied without pseudocount. Try ",
-             "hellinger with pseudocount = 1 or other numeric value.",
+             "transformation is being applied without (suitable) pseudocount. ",
+             "Try hellinger with pseudocount = 1 or other numeric value.",
```

Comment:
Does this apply pseudocount = 1 on relative abundances - this would be suitable for counts but not relative abundances.

Reply:
You can add pseudocount of `-1`. That would however not be suitable in some cases.

Reply:
Same comment than below. The default pseudocount choice should be different for counts and for relabundances. Hellinger should also yield identical result for both of these by default.